astropy/units/utils.py

```diff
@@ -229,32 +231,56 @@ def maybe_simple_fraction(p, max_denominator=100):
 
 
 def validate_power(p):
-    """Convert a power to a floating point value, an integer, or a Fraction.
+    """Check that a power can be converted to a floating point value.
+
+    Parameters
+    ----------
+    p : numerical
+        Power to be converted
+
+    Raises
+    ------
+    ValueError
+        If the power is an array in which not all elements are equal.
+    """
+    if p.__class__ is int or p.__class__ is Fraction:
+        return p
+    try:
+        float(p)
+    except Exception:
+        p = np.asanyarray(p)
+        if ((first := p.flat[0]) == p).all():
+            # All the same, now check it is OK.
+            float(first)
```

Comment:
shouldn't this recurse ?

Reply:
I had that first, but it then becomes an infinite loop, since if `first` is bad, it is turned again into an array (with one element), etc.